tools/CreateRuleFabricBot/CreateRuleFabricBot/Rules/PullRequestLabel/CodeOwnerEntry.cs

```diff
@@ -1,92 +1,112 @@
-            line = line.Trim();
-
-            if (string.IsNullOrEmpty(line) || line.StartsWith('#'))
+            get
             {
-                return null;
+                return !string.IsNullOrEmpty(PathExpression);
             }
+        }
 
-            List<string> entries = SplitLine(line).ToList();
+        private static string[] SplitLine(string line, char splitOn)
+        {
+            return line.Split(splitOn, StringSplitOptions.RemoveEmptyEntries);
+        }
 
-            if (!entries.Any())
-            {
-                return null;
-            }
+        public override string ToString()
+        {
+            return $"HasWildcard:{ContainsWildcard} Expression:{PathExpression} Owners:{string.Join(',', Owners)}  Labels:{string.Join(',', Labels)}";
+        }
 
-            CodeOwnerEntry coe = new CodeOwnerEntry
+        public void ParseLabels(string line)
+        {
+            // Parse a line that looks like # PRLabel: %Label, %Label
+            if (line.IndexOf(LabelMoniker, StringComparison.OrdinalIgnoreCase) == -1)
             {
-                // the first entry is the path/regex
-                PathExpression = entries[0],
-                ContainsWildcard = entries[0].Contains('*')
-            };
+                return;
+            }
 
-            // remove the '/' from the path
-            if (coe.PathExpression.StartsWith("/"))
+            // If we don't have a ':', nothing to do
+            int colonPosition = line.IndexOf(':');
+            if (colonPosition == -1)
             {
-                coe.PathExpression = coe.PathExpression.Substring(1);
+                return;
             }
 
-            for (int i = 1; i < entries.Count; i++)
+            line = line.Substring(colonPosition + 1).Trim();
+            foreach (string label in SplitLine(line, LabelSeparator).ToList())
             {
-                string entry = entries[i].Trim();
-
-                if (entry[0] == '@')
+                if (!string.IsNullOrWhiteSpace(label))
                 {
-                    coe.Owners.Add(entry.Substring(1));
-                }
-
-                if (entry[0] == '%')
-                {
-                    coe.Labels.Add(entry.Substring(1));
+                    Labels.Add(label.Trim());
                 }
             }
-
-            return coe;
         }
 
-        private static IEnumerable<string> SplitLine(string line)
+        public void ParseOwnersAndPath(string line)
         {
-            // Split the line into segments that are delimited by '@', '%' and the end of the string.
+            if (string.IsNullOrEmpty(line) || line.StartsWith('#'))
+            {
+                return;
+            }
+
+            line = ParsePath(line);
 
-            int previousSplit = 0;
-            for (int i = 0; i < line.Length; i++)
+            foreach (string author in SplitLine(line, OwnerSeparator).ToList())
```

Comment:
While it is OK to parse the owners do we actually use them anywhere?

Reply:
We don't, but the previous code did that, and I have tests for it. Any concerns to keep it? We could re-use this code in other places as well.

Reply:
No real concern other then it is dead code.

Reply:
Given that we want to combine the two ways of parsing CODEOWNER files, having this code already here might make that easier.